src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = (this.length === 0
+      || this[0] === null) ? '' : `${this[0]}`;
+
+    for (let i = 1; i < this.length; i++) {
+      switch (this[i]) {
+        case null:
+        case undefined:
+          joinedString += `${separator}`;
+          break;
+        default:
+          joinedString += `${separator}` + `${this[i]}`;
+          break;
+      }
+    }
```

Comment:
I suggest getting rid of switch, let's use if block instead
I also suggest using for of loop here